client/landing/jetpack-cloud/sections/backups/main.jsx

```diff
@@ -3,95 +3,227 @@
  * Internal dependencies
  */
 import { emptyFilter } from 'state/activity-log/reducer';
-import { getBackupAttemptsForDate, getDailyBackupDeltas, getEventsInDailyBackup } from './utils';
+// import { getBackupAttemptsForDate, getDailyBackupDeltas, getEventsInDailyBackup } from './utils';
 import { getSelectedSiteId } from 'state/ui/selectors';
 import { getSitePurchases } from 'state/purchases/selectors';
+import getSiteGmtOffset from 'state/selectors/get-site-gmt-offset';
+import getSiteTimezoneValue from 'state/selectors/get-site-timezone-value';
+import { applySiteOffset } from 'lib/site/timezone';
 import { requestActivityLogs } from 'state/data-getters';
 import { withLocalizedMoment } from 'components/localized-moment';
-import BackupDelta from '../../components/backup-delta';
-import DailyBackupStatus from '../../components/daily-backup-status';
+//import BackupDelta from '../../components/backup-delta';
+//import DailyBackupStatus from '../../components/daily-backup-status';
 import DatePicker from '../../components/date-picker';
 import getRewindState from 'state/selectors/get-rewind-state';
 import getSelectedSiteSlug from 'state/ui/selectors/get-selected-site-slug';
 import QueryRewindState from 'components/data/query-rewind-state';
 import QuerySitePurchases from 'components/data/query-site-purchases';
+import QuerySiteSettings from 'components/data/query-site-settings'; // Required to get site time offset
+
+const INDEX_FORMAT = 'YYYYMMDD';
+
+const backupActivityNames = [
+	'rewind__backup_complete_full',
+	'rewind__backup_complete_initial',
+	'rewind__backup_error',
+];
 
 class BackupsPage extends Component {
-	constructor( props ) {
-		super( props );
-		this.state = {
-			selectedDateString: props.moment().toISOString( true ),
-		};
+	//todo: add the rest of the expected propTypes
+	static propTypes = {
+		logs: PropTypes.object,
+		loading: PropTypes.bool,
+		siteId: PropTypes.number,
+		siteSlug: PropTypes.string,
+		siteTimezone: PropTypes.string,
+		siteGmtOffset: PropTypes.number,
+	};
+
+	state = {
+		selectedDate: new Date(),
+		backupsOnSelectedDate: [],
+		indexedLog: {},
+		oldestDateAvailable: new Date(),
+	};
+
+	componentDidMount() {
+		this.createIndexedLog();
+	}
+
+	componentDidUpdate( prevProps ) {
+		if ( prevProps.logs !== this.props.logs ) {
+			this.createIndexedLog();
+		}
+		if ( prevProps.siteId !== this.props.siteId ) {
+			this.resetState();
+		}
 	}
 
-	dateChange = selectedDateString => this.setState( { selectedDateString } );
+	resetState() {
+		this.setState( {
+			selectedDate: new Date(),
+			backupsOnSelectedDate: [],
+			indexedLog: {},
+			oldestDateAvailable: new Date(),
+		} );
+	}
 
-	hasRealtimeBackups = () =>
-		this.props.sitePurchases &&
-		!! this.props.sitePurchases.filter(
-			purchase => 'jetpack_backup_realtime' === purchase.productSlug
-		).length;
+	/**
+	 * Create an indexed log of backups based on the date of the backup and in the site time zone
+	 */
+	createIndexedLog() {
```

Comment:
I like the idea of creating an index with the dates! However, the way they are [placed into state reminds me of a lot of `getDerivedState`, and the article You Probably Don't Need Derived State](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#when-to-use-derived-state).

I feel like this method is here because of the access to the localized moment, but it actual has very little to do with the state of the component ( for example changing the selected date does not change anything about the index ). I would think this fits in much better in the `connect` function since it is basically another property ( retrieve the logs and transform them into the index in the connect function, then only pass the index and oldest date as properties )

Reply:
I'm going to move the method outside of the class and those fields on the state to the props